docs/abi.rst

```diff
@@ -0,0 +1,560 @@
+
+These ARC-4 types are not yet supported in PyTeal:
+
+* Non-power-of-2 unsigned integers under 64 bits, i.e. :code:`uint24`, :code:`uint48`, :code:`uint56`
+* Unsigned integers larger than 64 bits
+* Fixed point unsigned integers, i.e. :code:`ufixed<N>x<M>`
+
+Usage
+"""""""""""""""""""""
+
+Setting Values
+''''''''''''''''
+
+All basic types have a :code:`set()` method which can be used to assign a value. The arguments for this method differ depending on the ABI type. For convenience, here are links to the docs for each class's method:
+
+* :any:`abi.Uint.set()`, which is used by all :code:`abi.Uint` classes and :code:`abi.Byte`
+* :any:`abi.Bool.set()`
+* :any:`abi.StaticArray.set()`
+* :any:`abi.Address.set()`
+* :any:`abi.DynamicArray.set()`
+* :any:`abi.String.set()`
+* :any:`abi.Tuple.set()`
+
+A brief example is below. Please consult the documentation linked above for each method to learn more about specific usage and behavior.
+
+.. code-block:: python
+
+    myAddress = abi.make(abi.Address)
+    myBool = abi.make(abi.Bool)
+    myUint64 = abi.make(abi.Uint64)
+    myTuple = abi.make(abi.Tuple3[abi.Address, abi.Bool, abi.Uint64])
+
+    program = Seq(
+        myAddress.set(Txn.sender()),
+        myBool.set(Txn.fee() == Int(0)),
+        myUint64.set(5000),
+        myTuple.set(myAddress, myBool, myUint64)
+    )
+
+Getting Values
+''''''''''''''''''''''
+
+All basic types that represent a single value have a :code:`get()` method, which can be used to extract that value. The supported types and methods are:
+
+* :any:`abi.Uint.get()`, which is used by all :code:`abi.Uint` classes and :code:`abi.Byte`
+* :any:`abi.Bool.get()`
+* :any:`abi.Address.get()`
+* :any:`abi.String.get()`
+
+A brief example is below. Please consult the documentation linked above for each method to learn more about specific usage and behavior.
+
+.. code-block:: python
+
+    @Subroutine(TealType.uint64)
+    def minimum(a: abi.Uint64, b: abi.Uint64) -> Expr:
+        """Return the minimum value of the two arguments."""
+        return (
+            If(a.get() < b.get())
+            .Then(a.get())
+            .Else(b.get())
+        )
+
+Getting Values at Indexes
+''''''''''''''''''''''''''
+
+The types :code:`abi.StaticArray`, :code:`abi.Address`, :code:`abi.DynamicArray`, :code:`abi.String`, and :code:`abi.Tuple` are compound types, meaning they contain other types whose values can be extracted. The :code:`__getitem__` method, accessible by using square brackets to "index into" an object, can be used to extract these values.
+
+The supported methods are:
+
+* :any:`abi.StaticArray.__getitem__`, used for :code:`abi.StaticArray` and :code:`abi.Address`
+* :any:`abi.Array.__getitem__`, used for :code:`abi.DynamicArray` and :code:`abi.String`
+* :any:`abi.Tuple.__getitem__`
+
+Be aware that these methods return a :code:`ComputedValue`, TODO link to Computed Value section
```

Comment:
I'm sure there's a good reason for this, but as a user it's unintuitive that it wouldn't just return a value of the type in the array/tuple. So good callout, and I suggest we give a brief explanation for why this is the case